File: CreationalPattern/AbstractFactoryPattern/Java/Plants/PeaShooter.java

package Plants;

import Interface.Plant;

/**
 * Class for Peashooter
 */
public class PeaShooter implements Plant {
    // Var params for Peashooter
    private String name;
    private int hp;
    private int damage;
    private int price;
    
    // Constructor
    public PeaShooter() {
        this.name = "Peashooter";
        this.hp = 100;
        this.damage = 20;
        this.price = 100;
    }

    public PeaShooter(String name, int hp, int damage, int price) {
        this.name = name;
        this.hp = hp;
        this.damage = damage;
        this.price = price;
    }
    
    // Override method Display from Plant
    @Override
    public void display() {
        System.out.println("Name: " + name + ", Price: " + price + ", HP: " + hp + ", Damage: " + damage);
    }

    // Override method GetHealth from Plant
    @Override
    public int getHealth() {
        return hp;
    }

    // Getter and setter
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    public int getDamage() {
        return damage;
    }

    public void setDamage(int damage) {
        this.damage = damage;
    }

    public void setHp(int hp) {
        this.hp = hp;
    }

    @Override
    public void show() {
        // TODO Auto-generated method stub
        throw new UnsupportedOperationException("Unimplemented method 'show'");
    }
}

File: CreationalPattern/FactoryPattern/Golang/Plants/PeaShooter.go

package Plants

/**
 * PEASHOOTER struct, implement Plant interface
 */
type PeaShooter struct {
	Name  string
	Price int
	// TODO add more attribute ...
}

// Constructor for PeaShooter
func NewPeaShooter(params...interface{}) *PeaShooter {
	p := &PeaShooter {
		Name: "PeaShooter",
		Price: 100,
	}

	len := len(params)

	if (len > 0) {
		p.Name = params[0].(string)
	}

	if (len > 1) {
		p.Price = params[1].(int)
	}

	return p
}

func (s *PeaShooter) PlantName() string {
	return s.Name
}

func (s *PeaShooter) PlantPrice() int {
	return s.Price
}